pkg/collector/check/id.go

```diff
@@ -18,16 +18,16 @@ type ID string
 
 // Identify returns an unique ID for a check and its configuration
 func Identify(check Check, instance integration.Data, initConfig integration.Data) ID {
-	return BuildID(check.String(), instance, initConfig)
+	return BuildID(check.String(), check.ExtraString(), instance, initConfig)
 }
 
 // BuildID returns an unique ID for a check name and its configuration
-func BuildID(checkName string, instance, initConfig integration.Data) ID {
+func BuildID(checkName string, extraID string, instance, initConfig integration.Data) ID {
 	h := fnv.New64()
 	h.Write([]byte(instance))
 	h.Write([]byte(initConfig))
 
-	id := fmt.Sprintf("%s:%x", checkName, h.Sum64())
+	id := fmt.Sprintf("%s:%x %s", checkName, h.Sum64(), extraID)
```

Comment:
As the extraID is optional, we probably don't want a trailing space here when it's empty

Reply:
Good point, will change this.